### PR TITLE
added a few more examples. added a poll_some method which exposes the nitems parameter

### PR DESCRIPTION
**Pull request by dysn**

### CODE REVIEW

**Comment by rotty** [comment]
Thanks for the multiple examples, and sorry for not getting back to you sooner! I'll do a more detailed review of your commits, but two things up front:

- The `poll_some` change is not needed, I think -- you can use rust's slicing syntax to pass in a part of the poll item array, e.g., `zmq::poll(&mut items[0..1], -1)`
- I'd prefer pull requests not containing merge commits, as it clutters git history -- could you rebase on current master, also eliminating the `poll_some` change?

**Comment by zwb-ict** [comment]
@dysn Have you test syncpub and syncsub passed?  Please look at [#192](https://github.com/erickt/rust-zmq/issues/192) and give me a hand.

File: src/lib.rs [hunk 1]
```diff
@@ -1003,15 +1002,31 @@ impl<'a> PollItem<'a> {
 ///
 /// The result, if not `Err`, indicates the number of poll items that have
 /// events signaled.
-pub fn poll(items: &mut [PollItem], timeout: i64) -> Result<i32> {
+pub fn poll_some(items: &mut [PollItem], nitems: u32, timeout: i64) -> Result<i32> {
```
Comment (birkenfeld) [inline src/lib.rs]: Is it intended to have this used from outside? I don't see the advantage over `poll(&items[..nitems])`.

If it's going to be kept, the `nitems` parameter should at least be made a usize.

**Comment by rotty** [comment]
I've now integrated your work, thanks for your contribution! I've taken the liberty to apply the following changes to your PR:

- Remove your `poll_some` commit, as that method is superfluous (as already noted by me and @birkenfeld ).
- Eliminate the merge request -- please always submit pull requests without merge commits (i.e., rebased).
- Run the new code through rustfmt, as there was quite a bit of inconsistent use of whitespace in them.

**Pull request closed without merging**